azurerm/internal/services/videoanalyzer/video_analyzer_resource.go

```diff
@@ -0,0 +1,321 @@
+		},
+	}
+}
+
+func resourceVideoAnalyzerCreateUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).VideoAnalyzer.VideoAnalyzersClient
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	resourceId := parse.NewVideoAnalyzerID(subscriptionId, d.Get("resource_group_name").(string), d.Get("name").(string))
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceId.ResourceGroup, resourceId.Name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("checking for existing %s: %+v", resourceId, err)
+			}
+		}
+
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return tf.ImportAsExistsError("azurerm_video_analyzer", resourceId.ID())
+		}
+	}
+
+	identity, err := expandAzureRmVideoAnalyzerIdentity(d)
+	if err != nil {
+		return err
+	}
+	parameters := videoanalyzer.Model{
+		PropertiesType: &videoanalyzer.PropertiesType{
+			StorageAccounts: expandVideoAnalyzerStorageAccounts(d),
+		},
+		Location: utils.String(azure.NormalizeLocation(d.Get("location").(string))),
+		Identity: identity,
+		Tags:     tags.Expand(d.Get("tags").(map[string]interface{})),
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, resourceId.ResourceGroup, resourceId.Name, parameters); err != nil {
+		return fmt.Errorf("creating %s: %+v", resourceId, err)
+	}
+
+	d.SetId(resourceId.ID())
+	return resourceVideoAnalyzerRead(d, meta)
+}
+
+func resourceVideoAnalyzerRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).VideoAnalyzer.VideoAnalyzersClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.VideoAnalyzerID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Video Analyzer %q was not found in Resource Group %q - removing from state", id.Name, id.ResourceGroup)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("retrieving Video Analyzer %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+	}
+
+	d.Set("name", id.Name)
+	d.Set("resource_group_name", id.ResourceGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+
+	props := resp.PropertiesType
+	if props != nil {
+		accounts := flattenVideoAnalyzerStorageAccounts(props.StorageAccounts)
+		if err := d.Set("storage_account", accounts); err != nil {
+			return fmt.Errorf("flattening `storage_account`: %s", err)
+		}
+	}
+
+	flattenedIdentity, err := flattenAzureRmVideoServiceIdentity(resp.Identity)
+	if err != nil {
+		return fmt.Errorf("flattening `identity`: %s", err)
+	}
+
+	if err := d.Set("identity", flattenedIdentity); err != nil {
+		return fmt.Errorf("flattening `identity`: %s", err)
```

Comment:
technically this is setting, we're flattening above :)

```suggestion
		return fmt.Errorf("setting `identity`: %s", err)
```